packages/framer-motion/src/animation/generators/__tests__/utils.ts

```diff
@@ -0,0 +1,35 @@
+import { KeyframeGenerator } from "../types"
+
+export interface KeyframesMetadata {
+    keyframes: Array<string | number>
+    duration: number
+}
+
+const timeStep = 10
+const maxDuration = 10000
+export function pregenerateKeyframes(
+    generator: KeyframeGenerator<number>
+): KeyframesMetadata {
+    let timestamp = timeStep
+    let state = generator.next(0)
+    const keyframes: Array<string | number> = [state.value]
+
+    while (!state.done && timestamp < maxDuration) {
+        state = generator.next(timestamp)
+        keyframes.push(state.value)
+        timestamp += timeStep
+    }
+
+    const duration = timestamp - timeStep
+
+    /**
+     * If generating an animation that didn't actually move,
+     * generate a second keyframe so we have an origin and target.
+     */
+    if (keyframes.length === 1) keyframes.push(state.value)
+
+    return {
+        keyframes,
+        duration: duration / 1000,
```

Comment:
are there times when we're only interested in the duration? Because we could skip the modifying of the array, leading to a potential performance win.

Reply:
This one is just a utility function for tests and I don't think it often only returns `duration` however you've reminded me that we do have two similar functions that we might be able to unify and I'll keep this in mind if it's only being called for duration calculations